lib/client-assets.php

```diff
@@ -1459,6 +1459,52 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 		}
 	}
 
+	// Lock settings.
+	$user_id = wp_check_post_lock( $post->ID );
+	if ( $user_id ) {
+		/**
+		 * Filters whether to show the post locked dialog.
+		 *
+		 * Returning a falsey value to the filter will short-circuit displaying the dialog.
+		 *
+		 * @since 3.6.0
+		 *
+		 * @param bool         $display Whether to display the dialog. Default true.
+		 * @param WP_Post      $post    Post object.
+		 * @param WP_User|bool $user    The user id currently editing the post.
+		 */
+		if ( apply_filters( 'show_post_locked_dialog', true, $post, $user_id ) ) {
+			$locked = true;
+		}
+
+		$user_details = null;
+		if ( $locked ) {
+			$user         = get_userdata( $user_id );
```

Comment:
After some exploration, I'm going to keep this because there are some other places of the lock object where we don't have the id and we have just the avatar, so for consistency, I'll keep it this way for now.